astropy/utils/console.py

```diff
@@ -899,73 +899,80 @@ def __init__(self, msg, color='default', file=None, step=1,
         self._msg = msg
         self._color = color
         self._file = file
-        self._step = step
+        self._step = step + 1
+        self._cur_step = 0
         if chars is None:
             if conf.unicode_output:
                 chars = self._default_unicode_chars
             else:
                 chars = self._default_ascii_chars
         self._chars = chars
-
+        self._cur_char = 0
         self._silent = not isatty(file)
 
+        if self._silent:
+            self._iter = self._silent_iterator
+        else:
+            self._iter = self._iterator
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        file = self._file
+        write = file.write
+        flush = file.flush
+
+        if not self._silent:
+            write('\r')
+            color_print(self._msg, self._color, file=file, end='')
+        if exc_type is None:
+            color_print(' [Done]', 'green', file=file)
+        else:
+            color_print(' [Failed]', 'red', file=file)
+        flush()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        self._iter()
+
     def _iterator(self):
+        """
+        Update the spin wheel.
+        """
         chars = self._chars
-        index = 0
         file = self._file
         write = file.write
-        flush = file.flush
         try_fallback = True
 
-        while True:
+        if self._cur_step % self._step == 0:
             write('\r')
             color_print(self._msg, self._color, file=file, end='')
             write(' ')
             try:
                 if try_fallback:
-                    write = _write_with_fallback(chars[index], write, file)
+                    write = _write_with_fallback(chars[self._cur_char], write, file)
                 else:
-                    write(chars[index])
+                    write(chars[self._cur_char])
             except UnicodeError:
                 # If even _write_with_fallback failed for any reason just give
                 # up on trying to use the unicode characters
                 chars = self._default_ascii_chars
-                write(chars[index])
+                write(chars[self._cur_char])
```

Comment:
Since this is no longer a generator, you should actually update `self._chars` on line 962 so that it reads `self._chars = self._default_ascii_chars`.  Otherwise it will keep trying to use the unicode chars on every loop.

Reply:
Good catch

Reply:
Should be good now